### PR TITLE
Nathan/kalman filter integration

### PR DESCRIPTION
**Pull request by ExtraGravity**

### CODE REVIEW

**Comment by NathanMeulen** [comment]
I've fixed the compile errors, but I'm not quite sure what the checks check, and the log is fairly obtuse to me. Help here would be appreciated.

File: Src/ParachutesControl.c [hunk 1]
```diff
@@ -8,7 +8,30 @@
 #include "FlightPhase.h"
 #include "Data.h"
 
-static int MONITOR_FOR_PARACHUTES_PERIOD = 1000;
+// Pressure at spaceport america in 100*millibars on May 27, 2018
+static const int SEA_LEVEL_PRESSURE = 101421.93903699999; //TODO: THIS NEEDS TO BE UPDATED AND RECORDED ON LAUNCH DAY
+static const int MAIN_DEPLOYMENT_ALTITUDE = 1000; //TODO: FIND OUT WHAT THIS IS SUPPOSED TO BE!!! Units in meters.
+<<<<<<< HEAD
```
Comment (ExtraGravity) [inline Src/ParachutesControl.c]: You got merge conflicts in here.

File: Src/ParachutesControl.c [hunk 2]
```diff
@@ -8,7 +8,30 @@
 #include "FlightPhase.h"
 #include "Data.h"
 
-static int MONITOR_FOR_PARACHUTES_PERIOD = 1000;
+// Pressure at spaceport america in 100*millibars on May 27, 2018
+static const int SEA_LEVEL_PRESSURE = 101421.93903699999; //TODO: THIS NEEDS TO BE UPDATED AND RECORDED ON LAUNCH DAY
+static const int MAIN_DEPLOYMENT_ALTITUDE = 1000; //TODO: FIND OUT WHAT THIS IS SUPPOSED TO BE!!! Units in meters.
```
Comment (ExtraGravity) [inline Src/ParachutesControl.c]: Xionyi has this, I'll ask him for it.

File: Src/ParachutesControl.c [hunk 3]
```diff
@@ -45,17 +68,94 @@ int32_t readPressure(BarometerData* data)
...
+    double dt
+)
 {
-    // TODO
-    positionVector[0] = 0;
-    positionVector[1] = 0;
-    positionVector[2] = 0;
+    struct KalmanStateVector newState;
+    
+    // TODO: Figure out what the conversion factor is.
```
Comment (ExtraGravity) [inline Src/ParachutesControl.c]: Hey, let's do this as if the 16G conversion was working as we expected. i.e. 2048 is 1G.

File: Src/ParachutesControl.c [hunk 4]
```diff
@@ -45,17 +68,94 @@ int32_t readPressure(BarometerData* data)
...
+
+/**
+ * Takes an old state vector and current state measurements and
+ * converts them into a prediction of the rocket's current state.
+ *
+ * Params:
+ *   oldState - (KalmanStateVector) Past altitude, velocity and acceleration
+ *   currentAccel - (double) Measured acceleration
+ *   currentAltitude - (double) Measured altitude
+ *   dt - (double) Time since last step
```
Comment (ExtraGravity) [inline Src/ParachutesControl.c]: Hey is this in milliseconds? That's what it's coming in as.

Comment (NathanMeulen) [reply Src/ParachutesControl.c]: Thank you for reminding me. I'll convert this too.

File: Src/ParachutesControl.c [hunk 5]
```diff
@@ -8,7 +8,23 @@
 #include "FlightPhase.h"
 #include "Data.h"
 
-static int MONITOR_FOR_PARACHUTES_PERIOD = 1000;
+// Pressure at spaceport america in 100*millibars on May 27, 2018
+static const int SEA_LEVEL_PRESSURE = 101421.93903699999; //TODO: THIS NEEDS TO BE UPDATED AND RECORDED ON LAUNCH DAY
+static const int MAIN_DEPLOYMENT_ALTITUDE = 4572; // Units in meters. Equivalent of 15000 ft.
```
Comment (ExtraGravity) [inline Src/ParachutesControl.c]: Hey is this 15000 ft above ground level or sea level? It should be above ground level.

Comment (NathanMeulen) [reply Src/ParachutesControl.c]: Is it bad if we just hard code this? It's the easiest and most reliable option at the moment.

**Pull request merged**